File: 1777-determine-if-two-strings-are-close/1777-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        if(word1.length() != word2.length()){
            return false;
        }
        unordered_map<int,char>mp1;
        unordered_map<int,char>mp2;
        for(char ch:word1){
            mp1[ch]++;
        }
        for(char ch:word2){
            mp2[ch]++;
        }
        set<char>s1;
        set<char>s2;
        for(auto &x:mp1){
            s1.insert(x.first);
        }
        for(auto &x:mp2){
            s2.insert(x.first);
        }
        if(s1 != s2){
            return false;
        }
        vector<int>temp1;
        vector<int>temp2;
        for(auto &x:mp1){
            temp1.push_back(x.second);
        }
        for(auto &x:mp2){
            temp2.push_back(x.second);
        }
        sort(temp1.begin(),temp1.end());
        sort(temp2.begin(),temp2.end());
        
        return temp1 == temp2;
    }
};

File: 1777-determine-if-two-strings-are-close/1777-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False
        from collections import Counter as C
        mp1 = C(word1)
        mp2 = C(word2)

        if set(mp1.keys()) != set(mp2.keys()):
            return False
        if sorted(mp1.values()) != sorted(mp2.values()):
            return False
        return True